kernel/src/main/java/org/kframework/compile/AddSortInjections.java

```diff
@@ -147,7 +147,7 @@ private K internalAddSortInjections(K term, Sort expectedSort) {
                         }
                     }
                 }
-                throw new AssertionError();
+                return KApply(KLabel("inj", actualSort, expectedSort), KList(visitChildren(term, actualSort)), Att.empty().add(Sort.class, expectedSort));
             } else {
                 return KApply(KLabel("inj", actualSort, expectedSort), KList(visitChildren(term, actualSort)), Att.empty().add(Sort.class, expectedSort));
```

Comment:
I think we can drop the `else` branch and hoist these identical `return KApply` to